docs/docs/prompts/structured-prompts.md

```diff
@@ -0,0 +1,245 @@
+# Structured prompts
+
+Koog uses the type-safe Kotlin DSL to create structured prompts with control over message types,
+their order, and content.
+
+The structured prompts let you pre-configure conversation history with multiple messages, provide multimodal content, 
+examples, tool calls, and their results.
+
+## Basic structure
+
+The `prompt()` function creates a Prompt object with a unique ID and a list of messages:
+
+<!--- INCLUDE
+import ai.koog.prompt.dsl.prompt
+-->
+```kotlin
+val prompt = prompt("unique_prompt_id") {
+    // List of messages
+}
+```
+<!--- KNIT example-structured-prompts-01.kt -->
+
+## Message types
+
+The Kotlin DSL supports the following types of messages, each of which corresponds to a specific role in a conversation:
```

Comment:
There's also "Reasoning message" that represents "thinking steps" for models that support these: https://github.com/JetBrains/koog/blob/91ec42151a9c223bf97dbff0a5a18a8a70b29c64/prompt/prompt-model/src/commonMain/kotlin/ai/koog/prompt/message/Message.kt#L178